src/ln.js

```diff
@@ -47,7 +47,7 @@ function _ln(options, source, dest) {
   if (options.symlink) {
     fs.symlinkSync(source, dest, os.platform() === "win32" ? "junction" : null);
```

Comment:
We should only use `"junction"` if we know the source is a directory. Gonna handle this separately since we never test symlinking for directories.